htdocs/public/payment/newpayment.php

```diff
@@ -355,6 +397,10 @@
 		$urlok = preg_replace('/securekey=[^&]+/', '', $urlok);
 		$urlko = preg_replace('/securekey=[^&]+/', '', $urlko);
 
+		if ($source == 'conferencesubscription') {
+			$urlok .= '&invoice='.dol_encode($invoice->id, $dolibarr_main_instance_unique_id);
```

Comment:
There is no need to encrypt the invoice id here, we can set invoice=$invoice->id (dol_encode is useless since it is a reversable function).
But because we must protect from setting a bad invoiceid, it means the invoice id must be included into the generated securekey.

Reply:
Well, I used dol_hash as you wanted for the securekeys but I kept the encoded ids in the url. I can remove them if you really want, but isn't it better to keep them hidden ? Even if it is not that useful, at least people won't be able to know easily the ids.

Reply:
(and I removed this encoded invoice id because I noticed I wasn't using it anyway)

Reply:
We need a security for the global url and it must hash (one side cryptine) and not encode (2 direction encode/decode), the id of data we will pay, inside the url) but it is already set by
$encodedsecurekey = dol_hash($conf->global->EVENTORGANIZATION_SECUREKEY.'conferenceorbooth'.$project->id, 2);
If we need to include invoiceid, it must be both as a clear parmaeter, and included into this dol_hash. But if we don't need it, it is better to have it nowhere.

Can you fix the conflict ?

Reply:
Okay, I will fix it tomorrow, thank you for the explanations.

Reply:
Is it ok now @eldy ?